modules/currency-l0/src/main/scala/org/tessellation/currency/modules/Programs.scala

```diff
@@ -1,30 +1,39 @@
 package org.tessellation.currency.modules
 
 import cats.effect.Async
+import cats.effect.std.Random
 
-import org.tessellation.sdk.domain.cluster.programs.{Joining, PeerDiscovery}
+import org.tessellation.currency.http.P2PClient
+import org.tessellation.sdk.domain.cluster.programs.{Joining, L0PeerDiscovery, PeerDiscovery}
 import org.tessellation.sdk.infrastructure.snapshot.programs.Download
 import org.tessellation.sdk.modules.SdkPrograms
 
 object Programs {
 
-  def make[F[_]: Async](
+  def make[F[_]: Async: Random](
     sdkPrograms: SdkPrograms[F],
     storages: Storages[F],
-    services: Services[F]
+    services: Services[F],
+    p2pClient: P2PClient[F]
   ): Programs[F] = {
     val download = Download
       .make(
         storages.node,
         services.consensus
       )
 
-    new Programs[F](sdkPrograms.peerDiscovery, sdkPrograms.joining, download) {}
+    val l0PeerDiscovery = L0PeerDiscovery.make(
+      p2pClient.l0Cluster,
+      storages.l0Cluster
+    )
+
+    new Programs[F](sdkPrograms.peerDiscovery, l0PeerDiscovery, sdkPrograms.joining, download) {}
   }
 }
 
 sealed abstract class Programs[F[_]] private (
   val peerDiscovery: PeerDiscovery[F],
+  val l0PeerDiscovery: L0PeerDiscovery[F],
```

Comment:
I guess you could call it `globalL0PeerDiscovery` to help a little in identifying what it is actually (differentiating it from `peerDiscovery` which runs for other L0 currency peers).